spec/system/user_show_spec.rb

```diff
@@ -0,0 +1,78 @@
+require 'rails_helper'
+
+RSpec.describe 'user profile page', type: :system do
+  before do
+    @user1 = User.create(Name: 'Test User1',
+                         Photo: 'https://unsplash.com/photos/F_-0BxGuVvo',
+                         Bio: 'Test bio 1.',
+                         PostsCounter: 0)
+    @user2 = User.create(Name: 'Test User2',
+                         Photo: 'https://unsplash.com/photos/F_-0BxGuVvo',
+                         Bio: 'Test bio 2.',
+                         PostsCounter: 0)
+
+    @post1 = Post.create(author: @user1, Title: 'Post 1',
+                         Text: 'This is the content of the first post.',
+                         CommentsCounter: 0,
+                         LikesCounter: 0)
+    @post2 = Post.create(author: @user2, Title: 'Post 2',
+                         Text: 'This is the first post by Test User2.',
+                         CommentsCounter: 0, LikesCounter: 0)
+    @post3 = Post.create(author: @user1, Title: 'Post 3',
+                         Text: 'The second post is here with some interesting content.',
+                         CommentsCounter: 0, LikesCounter: 0)
+    @post4 = Post.create(author: @user2, Title: 'Post 4',
+                         Text: 'Another post by Test User2.',
+                         CommentsCounter: 0, LikesCounter: 0)
+    @post5 = Post.create(author: @user1, Title: 'Post 5',
+                         Text: 'Check out the third post with lots of exciting stuff.',
+                         CommentsCounter: 0, LikesCounter: 0)
+    @post6 = Post.create(author: @user2, Title: 'Post 6',
+                         Text: 'Yet another post by Test User2.',
+                         CommentsCounter: 0, LikesCounter: 0)
+
+    Comment.create(post: @post1, author: @user2, Text: 'I enjoyed reading this.')
+    Comment.create(post: @post2, author: @user1,
+                   Text: 'Looking forward to more posts!')
+  end
+
+  it "show the user's profile picture" do
+    visit "/users/#{@user1.id}"
+    expect(page).to have_selector("img[src*='#{@user1.Photo}']")
+    # sleep(2)
+  end
+
+  it "show the user's picture picture" do
+    visit "/users/#{@user1.id}"
+    expect(page).to have_content(@user1.Name)
+    # sleep(2)
+  end
+
+  it 'shows number of posts the user has written' do
+    visit "/users/#{@user1.id}"
+    expect(page).to have_content("Number of posts: #{@user1.PostsCounter}")
+    # sleep(2)
+  end
+
+  it 'shows theuser\'s bio ' do
+    visit "/users/#{@user1.id}"
+    expect(page).to have_content(@user1.Bio)
+    # sleep(2)
+  end
+
```

Comment:
- [ ]  Please make sure to write the below tests for the user show page as requested in the project requirements.
  - `Tests that asserts you can see a button that lets me view all of a user's posts`
  - `Tests that redirect to the user's post's index page when you click on a see all posts.`